From: =?utf-8?b?SMOpY3RvciBPcsOzbiBNYXJ0w61uZXo=?= <zumbi@debian.org>
Date: Wed, 20 Feb 2019 13:25:36 +0100
Subject: [RFC] Work around PR libc/13097 "linux-vdso.so.1" #2

Hi,

missed the x86_64-m32 case:

gdb/
2011-08-16  Jan Kratochvil  <jan.kratochvil@redhat.com>

	Work around PR libc/13097.
	* solib.c (update_solib_list): Ignore "linux-vdso.so.1".
---
 gdb/solib.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: gdb/gdb/solib.c
===================================================================
--- gdb.orig/gdb/solib.c
+++ gdb/gdb/solib.c
@@ -902,8 +902,11 @@ update_solib_list (int from_tty)
 
 	  try
 	    {
-	      /* Fill in the rest of the `struct so_list' node.  */
-	      if (!solib_map_sections (i))
+	      /* Fill in the rest of the `struct so_list' node.
+		 Work around PR libc/13097.  */
+	      if (!solib_map_sections (i)
+		  && strcmp (i->so_original_name, "linux-vdso.so.1") != 0
+		  && strcmp (i->so_original_name, "linux-gate.so.1") != 0)
 		{
 		  not_found++;
 		  if (not_found_filename == NULL)
